app/lib/features/news/widgets/news_full_view.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:acter/common/animations/like_animation.dart';
+import 'package:acter/features/news/providers/news_providers.dart';
+import 'package:acter/features/news/widgets/news_item/news_item.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class NewsFullView extends ConsumerStatefulWidget {
+  final List<NewsEntry> newsList;
+  final int initialPageIndex;
+
+  const NewsFullView({
+    super.key,
+    required this.newsList,
+    this.initialPageIndex = 0,
+  });
+
+  @override
+  ConsumerState<NewsFullView> createState() => _NewsVerticalViewState();
+}
+
+class _NewsVerticalViewState extends ConsumerState<NewsFullView> {
+  late PageController _pageController;
+
+  @override
+  void initState() {
+    super.initState();
+    _pageController = PageController(
+      initialPage: widget.initialPageIndex,
+    );
+  }
+
+  @override
+  void dispose() {
+    _pageController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return buildPagerView();
```

Comment:
Does it even need to be a method here??? The internal tree isn't subbed enough to consider it...

Reply:
Either way should be fine. I generally prefer to avoid much code directly in build method.